File: sic-api/Entities/Su/SuUserTask.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace sic_api.Entities.Su;

[Table("su_user_task")]
public class SuUserTask : BaseBusinessEntity
{
    [Required]
    [MaxLength(100)]
    [Column("title")]
    public string Title { get; set; } = default!;

    [Column("start_time")]
    public DateTime? StartTime { get; set; }

    [Column("end_time")]
    public DateTime? EndTime { get; set; }

    [MaxLength(2000)]
    [Column("description")]
    public string? Description { get; set; }

    [Required]
    [ForeignKey(nameof(Task))]
    [Column("task_id")]
    public Guid TaskId { get; set; }
    public SuTask Task { get; set; } = default!;
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuUserTask.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseBusinessEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "su_user_task")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuUserTask extends BaseBusinessEntity {

    @Column(name = "title", nullable = false, length = 100)
    private String title;

    @Column(name = "start_time")
    private LocalDateTime startTime;

    @Column(name = "end_time")
    private LocalDateTime endTime;

    @Column(name = "description", length = 2000)
    private String description;

    @Column(name = "task_id", insertable = false, updatable = false)
    private UUID taskId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "task_id", nullable = false)
    private SuTask task;
}